src/app/lists/EditModal.tsx

```diff
@@ -43,49 +43,34 @@ const DisplayContents = styled.div`
 `;
 
 const EditModal: React.FC<Props> = ({ data, onClose }) => {
-    const generateDefaultData = (): { [key: string]: string | null } => {
-        if (data !== null && data !== undefined) {
-            return { ...data.data, ...data.tooltips };
-        }
-        return {};
-    };
-
     const [toSubmit, setToSubmit] = useState<{ [key: string]: string | null }>(
-        generateDefaultData()
+        data !== null && data !== undefined ? { ...data.data, ...data.tooltips } : {}
     );
 
     const [errorMsg, setErrorMsg] = useState<string | null>(null);
 
-    // const headersAndTooltips: HeadersAndTooltips = [];
     const headersThatHaveTooltips = headers.filter(([key]) => key !== "item_name");
 
     const headersAndTooltips: HeadersAndTooltips = headersThatHaveTooltips.map((header, index) => {
         return [header, tooltips[index]];
     });
 
     const setKey = (event: React.ChangeEvent<HTMLInputElement>, key: string): void => {
-        const text = event.target.value as string;
+        const text = event.target.value;
         setToSubmit({ ...toSubmit, [key]: text });
     };
 
     const onSubmit = async (): Promise<void> => {
-        let operation;
-
-        if (data === null) {
-            operation = supabase.from("lists").insert(toSubmit);
-        } else {
-            operation = supabase
-                .from("lists")
-                .update(toSubmit)
-                .eq("primary_key", toSubmit.primary_key)
-                .select();
-        }
+        const table = supabase.from("lists");
+        const operation =
```

Comment:
Just curious: is there a reason for having a `const table` and a `const operation` and then `await operation` instead of just `await supabase.from ...`? Or is it a preference thing?

Reply:
I see, cool :)

Reply:
Personally disagree, I think this would be better if `await`s are attached to the asynchronous thing they are waiting for - e.g. in this example `data === null ? await table.insert(...) : await table.update(...)`
especially when you put the new lines in like this, it makes it very clear what's going on.
but I'm very happy to keep it like this! it's not an awful solution either way.